provider/cmd/pulumi-resource-aws-apigateway/apigateway/api.ts

```diff
@@ -1003,16 +991,6 @@ function addAuthorizersToSwagger(
       auth.authorizerName || `${swagger.info.title}-authorizer-${suffix}`;
     auth.authorizerName = authName;
 
-    // Check API authorizers - if its a new authorizer add it to the apiAuthorizers
-    // if the name already exists, we check that the authorizer references the same authorizer
-    if (!apiAuthorizers[authName]) {
```

Comment:
I worried about completely removing this validation. What is the behavior if there are two different authorizers that share the same name? Does one override the other?

Is there a way to do a partial check on the fields that that can be resolved? Or could we log a warning instead?

Reply:
@corymhall I spent some more time trying to make the equality checks work. I found that nested resources don't get properly resolved by `apply`. I was able to workaround this by constructing an object out of the CustomResource (Lambda function in this case).
The validation works now.